src/Service/MailTemplateHelper.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\MailTemplate;
+use App\Repository\MailTemplateMacroRepository;
+use App\Repository\MailTemplateRepository;
+use Doctrine\ORM\EntityManagerInterface;
+use PhpOffice\PhpWord\Element\AbstractElement;
+use PhpOffice\PhpWord\Element\TextRun;
+use PhpOffice\PhpWord\TemplateProcessor;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpClient\Exception\ClientException;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Mime\Part\DataPart;
+use Symfony\Component\Mime\Part\Multipart\FormDataPart;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Serializer\SerializerInterface;
+
+class MailTemplateHelper
+{
+    /**
+     * The options.
+     *
+     * @var array
+     */
+    private $options;
+
+    public function __construct(private MailTemplateRepository $mailTemplateRepository, private MailTemplateMacroRepository $mailTemplateMacroRepository, private EntityManagerInterface $entityManager, private SerializerInterface $serializer, private Filesystem $filesystem, array $mailTemplateHelperOptions)
+    {
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+        $this->options = $resolver->resolve($mailTemplateHelperOptions);
+    }
+
+    public function getMailTemplateTypeChoices(): array
+    {
+        return array_flip(array_map(static fn (array $spec) => $spec['label'], $this->options['template_types']));
+    }
+
+    /**
+     * Get entity for previewing a mail template.
+     *
+     * @return mixed The entity
+     *
+     * @throws \RuntimeException
+     */
+    public function getPreviewEntity(MailTemplate $mailTemplate)
+    {
+        $spec = $this->options['template_types'][$mailTemplate->getType()] ?? null;
+        if (isset($spec['entity_class_names']) && is_array($spec['entity_class_names'])) {
+            foreach ($spec['entity_class_names'] as $className) {
+                try {
+                    $repository = $this->entityManager->getRepository($className);
+                    $entity = $repository->findOneBy([]);
+                    if (null !== $entity) {
+                        return $entity;
+                    }
+                } catch (\Exception $exception) {
+                }
```

Comment:
Instead of using exception handling for flow control, you should consider just continuing the loop when no entity found? If you prefer to let the exception handling be the way to go maybe you should consider only catching relevant exceptions and let all other exceptions pass through.

Reply:
This is only for trying to get an entity for testing a template, but nevertheless I've improved the exception handling a bit in https://github.com/itk-dev/naevnssekretariatet/pull/61/commits/ad9b9c068958ea6fc7934a2daa7e4dd406cf946d.